Jellyfin.Data/Entities/Security/ApiKey.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Globalization;
+
+namespace Jellyfin.Data.Entities.Security
+{
+    /// <summary>
+    /// An entity representing an API key.
+    /// </summary>
+    public class ApiKey
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApiKey"/> class.
+        /// </summary>
+        /// <param name="name">The name.</param>
+        public ApiKey(string name)
+        {
+            Name = name;
+
+            AccessToken = Guid.NewGuid().ToString("N", CultureInfo.InvariantCulture);
+            DateCreated = DateTime.UtcNow;
+        }
+
+        /// <summary>
+        /// Gets the id.
+        /// </summary>
+        /// <remarks>
+        /// Identity, Indexed, Required.
+        /// </remarks>
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public int Id { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the date created.
+        /// </summary>
+        public DateTime DateCreated { get; set; }
+
+        /// <summary>
+        /// Gets or sets the date of last activity.
+        /// </summary>
+        public DateTime DateLastActivity { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name.
+        /// </summary>
+        [MaxLength(64)]
+        [StringLength(64)]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the access token.
+        /// </summary>
+        public string AccessToken { get; set; }
```

Comment:
Why not a guid?

Reply:
I originally had it as a Guid, but I changed it to a string to have greater flexibility if we wanted to change how they are generated for 11

Reply:
Isn't that what migrations are for?

Reply:
EF core migrations aren't great for changing the data type for a column without losing data, but in this case it doesn't matter because the underlying data type is text regardless. The API expects a string, so this results in the least amount of type conversion between the two.